dataprofiler/profilers/profile_builder.py

```diff
@@ -840,17 +857,25 @@ def _restore_data_labelers(self, data_labeler=None):
                         dirpath=data_labeler_options.data_labeler_dirpath,
                         load_options=None)
                 self.options.set(
-                    {'data_labeler.data_labeler_object': data_labeler})
+                    {'unstructured_options.data_labeler.data_labeler_object':
+                         data_labeler})
 
             except Exception as e:
                 utils.warn_on_profile('data_labeler', e)
-                self.options.set({'data_labeler.is_enabled': False})
+                self.options.set({'unstructured_options.data_labeler.is_enabled':
+                                      False})
 
         # Restore data labelers for unstructured data labeling
         if use_data_labeler:
             data_labeler_profile = self._profile._profiles['data_labeler']
             data_labeler_profile.data_labeler = data_labeler
 
+        # Restore structured data labeler if there was one
+        if data_labelers and data_labelers[1] and self.options and \
+                isinstance(self.options, ProfilerOptions):
+            self.options.structured_options.data_labeler.data_labeler_object \
+                = data_labelers[1]
```

Comment:
This will be fixed shortly, but originally it was the unstructured option's labeler in `0`, and the structured one in `1`, this will be fixed to be made more clear

Reply:
I think you do that, you should add a comment -- sounds good though.